RHEL6_7/system/initscripts/ifcfg/check

```diff
@@ -51,99 +51,115 @@ I case 3, to be completely correct you could check that user is not using rhel7
-		else:
-			if is_kernel(name):
-				if addr:
-					ethx_with_addr_count +=1
-				else:
-					log_slight_risk(full_path + " is old style ethX name without HWADDR, its name can change after the upgrade.")
-					warning = True
-			elif is_udev(name):
-				log_slight_risk(full_path + " variable DEVICE is similar to udev predictable network naming scheme. It might cause conflicts.")
-				warning = True
-
-		if ethx_with_addr_count > 1:
-			log_medium_risk("You use multiple network devices with old style 'ethX' names.")
-			warning = True
-		if ethx_with_addr_count == 1:
-			log_slight_risk("You use one network device with an old style 'ethX' name. This will work as long as you only have one, but it will break with multiple such devices.")
-			warning = True
-	return warning
+    ifcfg_path = "/etc/sysconfig/network-scripts/"
+    ifcfg_prefix = "ifcfg-"
+
+    def get_variable(path, variable):
+        err_msg = "Error while reading ifcfg scripts"
+        def err(output):
+            log_error(err_msg)
+            log_error(output.read())
+
+        try:
+            output = os.tmpfile();
+            (currentpath, _ ) = os.path.split(os.path.realpath(__file__))
+            return_code = subprocess.Popen([currentpath + "/get_var_by_name.sh", path, variable],
+            bufsize=0,
+            executable=None,
+            stdin=None,
+            stdout=output,
+            stderr=subprocess.STDOUT,
+            preexec_fn=None,
+            close_fds=True,
+            shell=False,
+            cwd=None,
+            env=None,
+            universal_newlines=False,
+            startupinfo=None,
+            creationflags=0).wait()
+        except:
+            output.seek(0)
+            err(output)
+            raise
+        output.seek(0)
+        if return_code < 0:#error
+            log_error(output.read())
+            raise err_msg
+        if return_code == 1:#undefined variable
+            return False
+        return output.read().strip()
+
+    def ls_scripts():
+        for f in [i for i in os.listdir(ifcfg_path) if i.startswith(ifcfg_prefix)]:
+            yield f
+    
+    def is_kernel(name):
+        return re.match("eth[0-9]+", name)
+
+    def is_udev(name):
+        if name[0:2] in ("en", "sl", "wl", "ww"):
+            if name[2] in ("b", "c", "o", "s", "x", "P", "p"):
+                return True
+        return False
+
+    def fix_ifnames():
+        script_result = subprocess.Popen("./preupg_network.sh", shell=True)
+        script_result.communicate()
+        exit_code = script_result.wait()
+        if exit_code == 1:
+            log_extreme_risk("The network interface names on the source system are in inconsistent state. Upgrade is not possible")
+        elif exit_code == 2:
+            log_extreme_risk("MACADDR directive found in the ifcfg files. Comment it out and rerun preupgrade-assistant Upgrade is not possible")
+        return True
```

Comment:
You can use just `return` or not call return at all. Having `return True` may be misleading to some because it indicates you return some specific value even though:
1) nobody consumes it
2) there is no other return, that would return something else, for example False